model/src/main/java/org/projectnessie/api/v2/params/ReferencesParams.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.api.v2.params;
+
+import javax.annotation.Nullable;
+import javax.ws.rs.QueryParam;
+import org.eclipse.microprofile.openapi.annotations.media.ExampleObject;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
+import org.immutables.builder.Builder.Constructor;
+import org.projectnessie.api.params.FetchOption;
+
+/**
+ * The purpose of this class is to include optional parameters that can be passed to API methods
+ * dealing with reference retrieval.
+ *
+ * <p>For easier usage of this class, there is {@link ReferencesParams#builder()}, which allows
+ * configuring/setting the different parameters.
+ */
+public class ReferencesParams extends AbstractParams<ReferencesParams> {
```

Comment:
This one can also include start+end key equivalents.

Reply:
#5591